pkg/doc/jwt/verifier.go

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright SecureKey Technologies Inc. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package jwt
+
+import (
+	"crypto"
+	"crypto/ed25519"
+	"crypto/rsa"
+	"errors"
+	"fmt"
+)
+
+// Verifier makes verification of JSON Web Token.
+type Verifier interface {
+
+	// Verify verifies JWT.
+	Verify(j JSONWebToken) error
+}
+
+// KeyResolver resolves public key based on what and kid.
+type KeyResolver interface {
```

Comment:
the new Crypto interface and default Tink implementation is already in place, see interface here:
https://github.com/hyperledger/aries-framework-go/blob/master/pkg/crypto/api.go

you can take advantage of that to Sign/Verify using KeyHandle (see https://github.com/hyperledger/aries-framework-go/blob/master/pkg/crypto/tinkcrypto/crypto_test.go#L107 for example of signing and verifying)

soon, the KMS implementation will also be pushed so we can take advantage of getting managed keys (without passing in raw key `[]byte`).

Reply:
I think we shouldn't be tightly bound to a KMS interface.

The document packages, JWT, JWE, etc. should take generic signer, encrypter, etc. interfaces and the framework should construct those from KMS.

This way the packages can be used independently from the rest of the framework but still take advantage of framework services like KMS.

Reply:
the Crypto api can also be split by operation (eg Signer, Verifier, Encrypter, Decrypter interfaces)